src/EnergyPlus/EMSManager.cc

```diff
@@ -1537,6 +1537,7 @@ namespace EMSManager {
         } // NumOfNodes > 0
 
         if (NumOutsideAirNodes > 0) {
+            DataGlobals::AnyLocalEnvironmentsInModel = true;
```

Comment:
@lgu1234 is this right?  Anytime you have outside air nodes you need to set this flag to true?  I thought this flag was only true if the simulation had any outside air nodes with specific heights, so that you needed to update the local environment nodes appropriately.  I feel like setting this true here could cause the simulation to attempt to update local environment nodes when it doesn't need to.  It'll probably be a minuscule performance effect, but more importantly it just doesn't seem right.

Could you clarify why you needed it here?

Reply:
AnyLocalEnvironmentsInModel is related to SurfaceProperty:LocalEnvironment and ZoneProperty:LocalEnvironment.

Reply:
I think it could be significant performance hit for places that look for surface properties to apply.  I don't remember for sure, but should search the code on  `AnyLocalEnvironmentsInModel` and see how it's used.

Reply:
Interesting.

Over in the [OA Node Manager](https://github.com/NREL/EnergyPlus/blob/eb68ccb2bfb3d01807412e33399299da8cdb4b11/src/EnergyPlus/OutAirNodeManager.cc), during the [GetInput for OutdoorAir:Node](https://github.com/NREL/EnergyPlus/blob/eb68ccb2bfb3d01807412e33399299da8cdb4b11/src/EnergyPlus/OutAirNodeManager.cc#L305), the flag is [set to true](https://github.com/NREL/EnergyPlus/blob/eb68ccb2bfb3d01807412e33399299da8cdb4b11/src/EnergyPlus/OutAirNodeManager.cc#L364-L367) for any node who has number of alphas greater than 1, which would imply that the field has scheduled conditions.  So the trigger there is not the height, but instead scheduled.  This makes more sense given the EMS connection here.

Outdoor air node actuators are always set up, and so with this change, every model that has outdoor air nodes will have a true value on this flag.  Or I'm still missing something.

@lgu1234 I'm going to push this toward the back burner for tonight, chime in in the morning and let's see if we can get this resolved.

Reply:
@Myoldmopar Your understanding is correct. The purpose to set AnyLocalEnvironmentsInModel=true is to allow all nodes defined under OutdoorAir:NodeList to be overridden by EMS.

Reply:
@Myoldmopar In addition, the AnyLocalEnvironmentsInModel is set to true in GetOutAIrNode Input by reading the "OutdoorAir:Node" object in OutAirNodeManager at line 365

                if (NumAlphas > 1) {
                    AnyLocalEnvironmentsInModel = true;
                }
My intention is to allow OutdoorAir:NodeList to have the same setting as OutdoorAir:Node.

Reply:
Since you have the check for `NumAlphas>1`, then why set it when if (NumOutsideAirNodes > 0`? It seems like we're using `AnyLocalEnvironmentsInModel` for dual purposes here, for local environment *and* for EMS.

Reply:
@mjwitte The check for NumAlphas>1 is the original code. The change I made is to mimic OutdorAir:Node for OutdoorAir:NodeList. I agree with you the AnyLocalEnvironmentsInModel is for dual purpose. We may need a new issue to cleanup later.

Reply:
If I'm following all of this, the only thing that the outdoor air node manager cares about is `IsLocalNode`. This can be true if there's a direct EMS actuator on the node or if one of the extended fields are used, correct?  Beyond that, I don't see why the outdoor air node manager or EMSmanager should set AnyLocalEnvironments at all. Only the LocalEnvironment objects should set that.

Reply:
Sorry for the piecemeal comments. So, this line in EMSManager should be deleted.

Reply:
@Myoldmopar @mjwitte I am going to delete this line